packages/engine/Specs/Scene/PrimitiveCollectionSpec.js

```diff
@@ -649,6 +649,42 @@ describe(
         primitives.lowerToBottom(p);
       }).toThrowDeveloperError();
     });
+
+    it("fires an event on primitive added", function () {
+      const primitiveAdded = primitives.primitiveAdded;
+
+      const onPrimitiveAddedSpy = jasmine.createSpy("event");
+      primitiveAdded.addEventListener(onPrimitiveAddedSpy);
+      expect(onPrimitiveAddedSpy).not.toHaveBeenCalled();
+      primitives.add(createLabels());
+      expect(onPrimitiveAddedSpy).toHaveBeenCalled();
```

Comment:
I would suggest using `toHaveBeenCalledWith` here and below to ensure the primitives were indeed passed to the callback.